### PR TITLE
feat(brew): add `brew shellenv` command at plugin load

### LINKED ISSUE #10389: Source homebrew shellenv to setup homebrew in new install location
### If the feature request is for a plugin or theme, specify it here.

brew

### If the feature solves a problem you have, specify it here.

Homebrew seems to install in /opt/homebrew now and has a shell script that needs to be eval'd to set it up.

### Describe the proposed feature.

Could the following snippet be included in the brew plugin to configure homebrew correctly.

```bash
# Homebrew
if [[ -f /opt/homebrew/bin/brew ]]; then
  eval "$(/opt/homebrew/bin/brew shellenv)"
fi
```

### Describe alternatives you've considered

I manually added the above snippet myself to my zshrc file

### Additional context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by carlosala**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

Title is quite descriptive 👍🏻 

Closes #10389

### CODE REVIEW

File: plugins/brew/brew.plugin.zsh [hunk 1]
```diff
@@ -1,11 +1,19 @@
+if (( $+commands[brew] )); then
```
Comment (mcornella) [inline plugins/brew/brew.plugin.zsh:1]: This check is backwards, it should be
```suggestion
if (( ! $+commands[brew] )); then
```

File: plugins/brew/brew.plugin.zsh [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+  return
+fi
+
+if [[ -z "$HOMEBREW_PREFIX" ]]; then
+  eval "$(brew shellenv)"
```
Comment (mcornella) [inline plugins/brew/brew.plugin.zsh:6]: I think we should support running brew by checking the absolute path to the executable, if the command is not found. 

That means rewriting the initial check and only bailing out if neither command is found (`/opt/homebrew/bin/brew` or `/usr/local/bin/brew`).

I just realised that Homebrew can also be installed in Linux, maybe we should also check for that (see linuxbrew).

Comment (carlosala) [reply plugins/brew/brew.plugin.zsh:6]: Hmm looks nice. I'll update it tomorrow 👌🏻

File: plugins/brew/brew.plugin.zsh (hunk 1 above)
Comment (carlosala) [reply plugins/brew/brew.plugin.zsh:1]: Ups sorry😅😅

**Comment by carlosala** [comment]
@mcornella it should be done! `brew shellenv` is editing the PATH, so it's not necessary to edit it manually in case of `brew` not being in the PATH 👌🏻

**Comment by carlosala** [comment]
@mcornella take a look on this asap! 😉

**Pull request merged**